### PR TITLE
[CAS-1273] Thread load more

### PR DESCRIPTION
**Pull request by ogkuzmin**
### 🎯 Goal

In order to merge `ChatDomain` into `ChatClient` we need to eliminate direct usage of ChatDomain and substitute it with the usage of `ChatClient`.
This PR substitutes usage of `ChatDomain` for the thread load more use case

### 🛠 Implementation details

1. Keep old approach if offline plugin isn't available (check toggle)
2. Use `ChatClient` to get more replies
3. Store a reference for `ThreadState` in message mode of VMs

### 🧪 Testing

Run apps and paginate thread (find or create thread with more than 30 replies) with enabled/disabled offline toggle

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [ ] <del> Changelog is updated with client-facing changes </del>
- [ ] <del> New code is covered by unit tests </del>
- [ ] <del> Comparison screenshots added for visual changes </del>
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatApp.kt [hunk 1]
```diff
@@ -23,16 +28,23 @@ class ChatApp : Application() {
...
 
+    @OptIn(InternalStreamChatApi::class)
+    private fun initializeToggleService() {
+        ToggleService.init(applicationContext, mapOf(ToggleService.TOGGLE_KEY_OFFLINE to BuildConfig.DEBUG))
+    }
+
+    @OptIn(ExperimentalStreamChatApi::class)
     private fun setupStreamSdk() {
-        val client = ChatClient.Builder("qx5us2v6xvmh", applicationContext)
+        val offlinePlugin = OfflinePlugin(Config(userPresence = true, persistenceEnabled = true))
```
Comment (ogkuzmin) [inline stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatApp.kt:42]: We need to use `OfflinePlugin` for debug build in compose sample to catch all issues before initial release of domain merge

**Review by amitkma — APPROVED** [review]
LGTM

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt [hunk 2]
```diff
@@ -435,15 +446,38 @@ public class MessageListViewModel(
...
 
+    /**
+     * Load older messages for the specified thread [MessageMode.MessageThread.parentMessage].
+     *
+     * @param threadMode Current thread mode.
+     */
+    private fun threadLoadMore(threadMode: MessageMode.MessageThread) {
+        threadMessagesState = threadMessagesState.copy(isLoadingMore = true)
+        if (ToggleService.isEnabled(ToggleService.TOGGLE_KEY_OFFLINE).not()) {
+            threadMessagesState = threadMessagesState.copy(isLoadingMore = true)
```
Comment (adasiewiczr) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt:464]: I think we can remove this line. We are already changing this state two lines above

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt [hunk 3]
```diff
@@ -435,15 +446,38 @@ public class MessageListViewModel(
         val messageMode = messageMode
 
         if (messageMode is MessageMode.MessageThread) {
-            threadMessagesState = threadMessagesState.copy(isLoadingMore = true)
-            chatDomain.threadLoadMore(channelId, messageMode.parentMessage.id, messageLimit)
```
Comment (adasiewiczr) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt:439]: Shouldn't we deprecate `chatDomain.threadLoadMore`?

Comment (amitkma) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt:439]: As Oleg said to me, we shouldn't deprecate as there is no public mechanism to thread load more until we release OfflinePlugin.

**Review by adasiewiczr — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by filbabic — COMMENTED** [review]
Looks good to me, works nicely! :]

Can we please add this to the changelog @amit @adasiewiczr 

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by filbabic — APPROVED** [review]

**Pull request merged**